sway/config/output.c

```diff
@@ -835,12 +837,13 @@ static bool _spawn_swaybg(char **command) {
 		sway_log_errno(SWAY_ERROR, "close failed");
 		return false;
 	}
-	if (waitpid(pid, NULL, 0) < 0) {
+	int fork_status = 0;
+	if (waitpid(pid, &fork_status, 0) < 0) {
 		sway_log_errno(SWAY_ERROR, "waitpid failed");
 		return false;
 	}
 
-	return true;
+	return WEXITSTATUS(fork_status) == EXIT_SUCCESS;
```

Comment:
I think that `WEXITSTATUS` can only be used if `WIFEXITED` returns true. Maybe we can add `WIFEXITED(fork_status) &&` before?

That's pretty much an edge case though.

Reply:
Didn't know about that, reading the documentation helps - thanks! :)